### PR TITLE
Add transactions

### PR DESCRIPTION
**Pull request by davazp**

### CODE REVIEW

**Comment by alextes** [comment]
Aaah, it was adding `got` that I was making optional.

**Comment by alextes** [comment]
Me and @tkers had trouble making Axios work and their docs are meh.

File: client/index.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+  async get(key) {
+    return getKey(this.request, undefined, key);
+  }
+
+  async writeTransaction() {
+    const response = await this.request.post("/begin-write-transaction");
+    const token = response.data;
+    return new WriteTransaction(this.request, token);
+  }
```
Comment (alextes) [inline client/index.js:29]: Why all the inheritance @davazp ? It's a hack project. I don't much care. Still, this function used to be a couple lines of code, now it's three levels of classes. If you'd like to relate the transactions to a connection, how about we pass the connection as an argument?

**Comment by alextes** [comment]
Alright, no more changes from me. Good to merge 👍 .

**Comment by alextes** [comment]
Actually, I guess I can merge it.

File: src/transactions.js [hunk 2]
```diff
@@ -96,6 +98,18 @@ function releaseWriteTransaction(tx) {
   release();
 }
 
+async function commitTransaction(tx) {
+  commitedTransactionIds = [...commitedTransactionIds, tx.txid];
+}
+
+function isCommited(txid) {
+  console.log(
```
Comment (tkers) [inline src/transactions.js:106]: Maybe remove this

**Pull request merged**